### PR TITLE
fix more warnings

### PR DESCRIPTION
**Pull request by adizaimi**
fix more warnings

### CODE REVIEW

File: db/sqlglue.c [hunk 1]
```diff
@@ -2233,8 +2217,8 @@ static void genid_hash_add(BtCursor *cur, int rrn, unsigned long long genid)
 
     /* printf("genid_hash_add(cur %d rrn %d genid %08llx rc %d)\n",
      * cur->cursorid, rrn, genid, rc); */
-    rc = bdb_temp_hash_insert(cur->bt->genid_hash, &key, sizeof(struct key),
-                              &genid, sizeof(genid));
```
Comment (nirbhayc) [inline db/sqlglue.c]: Does bdb_temp_hash_insert() log about failures? If not, may be we should check the return code and log an appropriate message?

File: db/sqlglue.c [hunk 2]
```diff
@@ -2672,15 +2643,14 @@ static int cursor_move_compressed(BtCursor *pCur, int *pRes, int how)
...
      * and if peer_dropped_connection is true, we'll get NO SQL ENGINE and
      * a wasted thread apparently.
-     rc = cursor_move_preprop(pCur, pRes, how, &done, 0);
-     if (done)
-     return rc;
+    int done = 0;
+    int rc = SQLITE_OK;
+    rc = cursor_move_preprop(pCur, pRes, how, &done, 0);
+    if (done)
+    return rc;
```
Comment (nirbhayc) [inline db/sqlglue.c]: Check indentation

File: db/sqlglue.c [hunk 3]
```diff
@@ -2672,15 +2643,14 @@ static int cursor_move_compressed(BtCursor *pCur, int *pRes, int how)
...
-    int rc = SQLITE_OK;
-
     /* skip preprop. if we're called from sqlite3_open_serial
      * and if peer_dropped_connection is true, we'll get NO SQL ENGINE and
      * a wasted thread apparently.
-     rc = cursor_move_preprop(pCur, pRes, how, &done, 0);
-     if (done)
-     return rc;
+    int done = 0;
+    int rc = SQLITE_OK;
```
Comment (nirbhayc) [inline db/sqlglue.c]: No point in initializing rc, gets overwritten below.

File: db/sqlglue.c [hunk 4]
```diff
@@ -5415,7 +5361,7 @@ int sqlite3BtreeMovetoUnpacked(BtCursor *pCur, /* The cursor to be moved */
     if (pCur->bt->is_temporary) {
         if (pIdxKey) {
             if (bdb_is_hashtable(pCur->tmptable->tbl)) {
-                Mem mem = {0};
+                Mem mem = {{0}};
```
Comment (nirbhayc) [inline db/sqlglue.c]: What was the warning? Did you check if this really works (or only suppresses the warning)?

File: db/sqlglue.c [hunk 5]
```diff
@@ -6410,7 +6352,7 @@ int get_data(BtCursor *pCur, struct schema *sc, uint8_t *in, int fnum, Mem *m,
 {
     int null;
     i64 ival;
-    double dval;
+    double dval = 0;
```
Comment (nirbhayc) [inline db/sqlglue.c]: IMO, we should move this to the specific switch case.

File: db/types.c [hunk 6]
```diff
@@ -9935,9 +9916,7 @@ static void decimal64_ondisk_to_double(server_decimal64_t *pdec64,
                                        decDouble *dn)
 {
     char decimals[DECDOUBLE_PACKED_COEF];
-    int exponent;
-    int sign;
-    int rc = 0;
+    int exponent = 0;
```
Comment (nirbhayc) [inline db/types.c]: Looks like exponent always gets a value, why would compiler complain about this one?

File: db/sqlglue.c [hunk 7]
```diff
@@ -1440,9 +1430,9 @@ char *sql_field_default_trans(struct field *f, int is_out)
     void *this_default;
     unsigned int this_default_len;
     struct field_conv_opts_tz outopts = {0};
-    unsigned long long uival;
-    long long ival;
-    double dval;
+    unsigned long long uival = 0;
+    long long ival = 0;
+    double dval = 0;
```
[the remark this answers is not in the document]
Comment (adizaimi) [reply db/sqlglue.c]: ok. doing so.

File: db/types.c (hunk 6 above)
Comment (adizaimi) [reply db/types.c]: I dont think in this case compiler is that smart:
/bb/comdb2a/adi/master_branch/db/types.c:9578:15: warning: ‘exponent’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         logmsg(LOGMSG_USER, "Dec64 unmake_order IN (exponent %d %x):\n", *exponent,

**Comment by adizaimi** [comment]
Thank you Nirbhay.

**Pull request merged**